catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.java

```diff
@@ -700,14 +700,17 @@ private void switchToCatblocks() {
 
 		int scriptIndex = -1;
 
-		int firstVisible = listView.getFirstVisiblePosition();
-
-		if (firstVisible >= 0) {
-			Object firstBrick = listView.getItemAtPosition(firstVisible);
-			if (firstBrick instanceof Brick) {
-				Script scriptOfBrick = ((Brick) firstBrick).getScript();
-				scriptIndex =
-						ProjectManager.getInstance().getCurrentSprite().getScriptIndex(scriptOfBrick);
+		if (ProjectManager.getInstance().getCurrentSprite().getScriptList() != null &&
+				ProjectManager.getInstance().getCurrentSprite().getScriptList().size() > 0) {
```

Comment:
This condition is a bit of an overhead. Just use `listView.getCount() > 0 && firstVisible >= 0` - this should actually suffice. No need to use `scriptList.size()` from `ProjectManager` here.

Reply:
Thanks - I didn't think of that. It's changed now.